javagen/src/main/java/com/azure/autorest/template/ModelTemplate.java

```diff
@@ -762,6 +792,52 @@ private void addModelConstructor(ClientModel model, JavaVisibility constructorVi
                 }
             }
         });
+
+        if (generatePrivateConstructorForJackson) {
+            addGeneratedAnnotation(classBlock);
+            classBlock.annotation("JsonCreator");
+
+            StringBuilder constructorPropertiesAsWireType =
+                    new StringBuilder(constructorPropertiesStringBuilderCapacity);
+
+            StringBuilder constructorPropertiesInvokePublicConstructor =
+                    new StringBuilder(constructorPropertiesStringBuilderCapacity);
+
+            final Consumer<ClientModelProperty> addParameterInvokePublicConstructor = p -> {
+                if (constructorPropertiesInvokePublicConstructor.length() > 0) {
+                    constructorPropertiesInvokePublicConstructor.append(", ");
+                }
+
+                if (p.getWireType() == p.getClientType()) {
+                    constructorPropertiesInvokePublicConstructor.append(p.getName());
+                } else {
+                    constructorPropertiesInvokePublicConstructor.append(p.getWireType().convertToClientType(p.getName()));
+                }
+            };
+
+            for (ClientModelProperty property : requiredParentProperties) {
+                if (constructorPropertiesAsWireType.length() > 0) {
+                    constructorPropertiesAsWireType.append(", ");
+                }
+
+                addModelConstructorParameterAsWireType(property, constructorPropertiesAsWireType);
+
+                addParameterInvokePublicConstructor.accept(property);
+            }
+            for (ClientModelProperty property : requiredProperties) {
+                if (constructorPropertiesAsWireType.length() > 0) {
+                    constructorPropertiesAsWireType.append(", ");
+                }
+
+                addModelConstructorParameterAsWireType(property, constructorPropertiesAsWireType);
+
+                addParameterInvokePublicConstructor.accept(property);
+            }
+
+            classBlock.privateConstructor(model.getName() + "(" + constructorPropertiesAsWireType + ")", constructor -> {
+                constructor.line("this(" + constructorPropertiesInvokePublicConstructor + ");");
+            });
+        }
```

Comment:
Main logic here, to generate another private ctor with parameters as wire types.